Recrutify/Recrutify.Services/Validators/UpdateProjectValidator.cs

```diff
@@ -1,14 +1,55 @@
-﻿using FluentValidation;
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentValidation;
+using Recrutify.DataAccess.Repositories.Abstract;
 using Recrutify.Services.DTOs;
 
 namespace Recrutify.Services.Validators
 {
     public class UpdateProjectValidator : BaseProjectValidator<UpdateProjectDTO>
     {
-        public UpdateProjectValidator()
+        private readonly IProjectRepository _projectRepository;
+
+        public UpdateProjectValidator(IProjectRepository projectRepository)
+        {
+            _projectRepository = projectRepository;
+
+            ConfigureRules();
+        }
+
+        private void ConfigureRules()
         {
+            RuleFor(p => p)
+                .CustomAsync(ProjectAreExistingAsync);
             RuleFor(p => p.Id)
                 .NotEmpty();
+            RuleFor(p => p.EndDate)
+                .NotNull()
+                .GreaterThanOrEqualTo(DateTime.UtcNow.Date)
+                .WithMessage("Date must be in the today or future!");
+        }
+
+        private async Task ProjectAreExistingAsync(UpdateProjectDTO dto, ValidationContext<UpdateProjectDTO> context, CancellationToken cancellationToken)
```

Comment:
You are checking here not existence of project. Rename to UmmutableFieldsAreKeepingUnchanged

Reply:
done